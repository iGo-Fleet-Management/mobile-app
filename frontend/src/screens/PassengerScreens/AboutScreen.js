import React from 'react';
import { View, Text, StyleSheet } from 'react-native';
import Header from '../../components/common/Header';
import { FontAwesome5 } from '@expo/vector-icons';
import { SafeAreaView } from 'react-native-safe-area-context';

const AboutScreen = ({ navigation }) => {
  return (
    <SafeAreaView style={styles.aboutContainer} edges={['top']}>
    <View style={styles.aboutContainer}>
      <Header 
        title="Sobre o iGO" 
        onArrowBackPress={() => navigation.goBack()}  
      />
      
      <View style={styles.aboutCard}>
        <Text style={styles.aboutTitle}>iGO</Text>
        <Text style={styles.aboutVersion}>Versão 1.0.0</Text>
        
        <Text style={styles.aboutDescription}>
          O iGO é um aplicativo de transporte desenvolvido para facilitar seus deslocamentos diários com 
          segurança e praticidade.
        </Text>
        
        <View style={styles.featureItem}>
          <FontAwesome5 name="map-marked-alt" size={22} color="#4285F4" />
          <Text style={styles.featureText}>Acompanhamento em tempo real</Text>
        </View>
        
        <Text style={styles.aboutFooter}>
          Desenvolvido como projeto acadêmico
        </Text>
      </View>
    </View>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  aboutContainer: {
    flex: 1,
    backgroundColor: '#f5f5f5',
  },
  aboutCard: {
    backgroundColor: '#ffffff',
    borderRadius: 8,
    padding: 20,
    margin: 15,
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.1,
    shadowRadius: 4,
    elevation: 3,
  },
  aboutTitle: {
    fontSize: 22,
    fontWeight: 'bold',
    textAlign: 'center',
    color: '#3f51b5',
  },
  aboutVersion: {
    fontSize: 14,
    color: '#757575',
    textAlign: 'center',
    marginBottom: 20,
  },
  aboutDescription: {
    fontSize: 15,
    lineHeight: 22,
    marginBottom: 20,
    textAlign: 'center',
  },
  featureItem: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: 15,
  },
  featureText: {
    marginLeft: 15,
    fontSize: 15,
  },
  aboutFooter: {
    marginTop: 30,
    textAlign: 'center',
    fontSize: 14,
    color: '#9e9e9e',
    fontStyle: 'italic',
  },
});

export default AboutScreen;