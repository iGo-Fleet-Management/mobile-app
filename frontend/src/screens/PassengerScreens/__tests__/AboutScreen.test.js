import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import AboutScreen from '../AboutScreen';
import { FontAwesome5 } from '@expo/vector-icons';

// Mock da navegação
const mockNavigation = {
  goBack: jest.fn(),
};

// Mock do Header
jest.mock('../../../components/common/Header', () => {
  const { View, Text, TouchableOpacity } = require('react-native');
  return ({ title, onArrowBackPress }) => (
    <View testID="header">
      <TouchableOpacity testID="backButton" onPress={onArrowBackPress}>
        <Text>{title}</Text>
      </TouchableOpacity>
    </View>
  );
});

// Mock dos ícones
jest.mock('@expo/vector-icons', () => {
  const { View } = require('react-native');
  return {
    FontAwesome5: jest.fn(({ name, size, color, testID }) => {
      return <View testID={testID || `icon-${name}`} />;
    }),
  };
});

describe('Tela Sobre', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('deve renderizar corretamente todos os elementos', () => {
    const { getByText, getByTestId } = render(
      <AboutScreen navigation={mockNavigation} />
    );

    expect(getByText('Sobre o iGO')).toBeTruthy();
    expect(getByText('iGO')).toBeTruthy();
    expect(getByText('Versão 1.0.0')).toBeTruthy();
    expect(getByText('Acompanhamento em tempo real')).toBeTruthy();
    expect(getByTestId('icon-map-marked-alt')).toBeTruthy();
  });

  it('deve aplicar os estilos corretamente', () => {
    const { getByText } = render(
      <AboutScreen navigation={mockNavigation} />
    );

    const title = getByText('iGO');
    expect(title).toHaveStyle({
      fontSize: 22,
      fontWeight: 'bold',
      color: '#3f51b5',
      textAlign: 'center'
    });

    const version = getByText('Versão 1.0.0');
    expect(version).toHaveStyle({
      fontSize: 14,
      color: '#757575',
      textAlign: 'center',
      marginBottom: 20
    });
  });

  it('deve voltar para tela anterior quando botão de voltar é pressionado', () => {
    const { getByTestId } = render(
      <AboutScreen navigation={mockNavigation} />
    );

    fireEvent.press(getByTestId('backButton'));
    expect(mockNavigation.goBack).toHaveBeenCalled();
  });

  it('deve exibir o item de feature com estilo correto', () => {
    const { getByText } = render(
      <AboutScreen navigation={mockNavigation} />
    );

    const featureItem = getByText('Acompanhamento em tempo real').parent;
    expect(featureItem).toHaveStyle({
      flexDirection: 'row',
      alignItems: 'center',
      marginBottom: 15
    });
  });

  it('deve corresponder ao snapshot', () => {
    const { toJSON } = render(
      <AboutScreen navigation={mockNavigation} />
    );
    expect(toJSON()).toMatchSnapshot();
  });
});