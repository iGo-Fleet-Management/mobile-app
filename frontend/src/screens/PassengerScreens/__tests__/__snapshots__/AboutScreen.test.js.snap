// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutSreen deve corresponder ao snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#f5f5f5",
      "flex": 1,
    }
  }
>
  <View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
      testID="backButton"
    >
      <Text>
        Sobre o iGO
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "backgroundColor": "#ffffff",
        "borderRadius": 8,
        "elevation": 3,
        "margin": 15,
        "padding": 20,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 4,
      }
    }
  >
    <Text
      style={
        {
          "color": "#3f51b5",
          "fontSize": 22,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      iGO
    </Text>
    <Text
      style={
        {
          "color": "#757575",
          "fontSize": 14,
          "marginBottom": 20,
          "textAlign": "center",
        }
      }
    >
      Versão 1.0.0
    </Text>
    <Text
      style={
        {
          "fontSize": 15,
          "lineHeight": 22,
          "marginBottom": 20,
          "textAlign": "center",
        }
      }
    >
      O iGO é um aplicativo de transporte desenvolvido para facilitar seus deslocamentos diários com segurança e praticidade.
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 15,
        }
      }
    >
      <View
        testID="icon-map-marked-alt"
      />
      <Text
        style={
          {
            "fontSize": 15,
            "marginLeft": 15,
          }
        }
      >
        Acompanhamento em tempo real
      </Text>
    </View>
    <Text
      style={
        {
          "color": "#9e9e9e",
          "fontSize": 14,
          "fontStyle": "italic",
          "marginTop": 30,
          "textAlign": "center",
        }
      }
    >
      Desenvolvido como projeto acadêmico
    </Text>
  </View>
</View>
`;

exports[`Tela Sobre deve corresponder ao snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#f5f5f5",
      "flex": 1,
    }
  }
>
  <View
    testID="header"
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
      testID="backButton"
    >
      <Text>
        Sobre o iGO
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "backgroundColor": "#ffffff",
        "borderRadius": 8,
        "elevation": 3,
        "margin": 15,
        "padding": 20,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 4,
      }
    }
  >
    <Text
      style={
        {
          "color": "#3f51b5",
          "fontSize": 22,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      iGO
    </Text>
    <Text
      style={
        {
          "color": "#757575",
          "fontSize": 14,
          "marginBottom": 20,
          "textAlign": "center",
        }
      }
    >
      Versão 1.0.0
    </Text>
    <Text
      style={
        {
          "fontSize": 15,
          "lineHeight": 22,
          "marginBottom": 20,
          "textAlign": "center",
        }
      }
    >
      O iGO é um aplicativo de transporte desenvolvido para facilitar seus deslocamentos diários com segurança e praticidade.
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 15,
        }
      }
    >
      <View
        testID="icon-map-marked-alt"
      />
      <Text
        style={
          {
            "fontSize": 15,
            "marginLeft": 15,
          }
        }
      >
        Acompanhamento em tempo real
      </Text>
    </View>
    <Text
      style={
        {
          "color": "#9e9e9e",
          "fontSize": 14,
          "fontStyle": "italic",
          "marginTop": 30,
          "textAlign": "center",
        }
      }
    >
      Desenvolvido como projeto acadêmico
    </Text>
  </View>
</View>
`;
